x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/search_index.tsx

```diff
@@ -80,24 +78,39 @@ export const SearchIndex: React.FC = () => {
     productAccess: { hasAppSearchAccess },
     productFeatures: { hasDefaultIngestPipeline },
   } = useValues(KibanaLogic);
-  const isAppGuideActive = useObservable(
-    guidedOnboarding.guidedOnboardingApi!.isGuideStepActive$('appSearch', 'add_data')
-  );
-  const isWebsiteGuideActive = useObservable(
-    guidedOnboarding.guidedOnboardingApi!.isGuideStepActive$('websiteSearch', 'add_data')
-  );
-  const isDatabaseGuideActive = useObservable(
-    guidedOnboarding.guidedOnboardingApi!.isGuideStepActive$('databaseSearch', 'add_data')
-  );
+
+  useEffect(() => {
+    const subscription = guidedOnboarding?.guidedOnboardingApi
+      ?.isGuideStepActive$('appSearch', 'add_data')
+      .subscribe((isStepActive) => {
+        if (isStepActive) {
```

Comment:
I think the check before also included `index.count`
```suggestion
        if (isStepActive && index?.count) {
```

Reply:
Nice catch! Fixed.